alibi_detect/utils/keops/kernels.py

```diff
@@ -115,13 +117,36 @@ def forward(self, x: LazyTensor, y: LazyTensor, infer_sigma: bool = False) -> La
             kernel_mat = kernel_mat.sum(-1) / len(self.sigma)
         return kernel_mat
 
+    def get_config(self) -> dict:
+        """
+        Returns a serializable config dict (excluding the input_sigma_fn, which is serialized in alibi_detect.saving).
+        """
+        cfg = self.config.copy()
+        if isinstance(cfg['sigma'], torch.Tensor):
+            cfg['sigma'] = cfg['sigma'].detach().cpu().numpy().tolist()
+        cfg.update({'flavour': Framework.KEOPS.value})
+        return cfg
+
+    @classmethod
+    def from_config(cls, config):
+        """
+        Instantiates a kernel from a config dictionary.
+
+        Parameters
+        ----------
+        config
+            A kernel config dictionary.
+        """
+        config.pop('flavour')
+        return cls(**config)
+
 
 class DeepKernel(nn.Module):
     def __init__(
         self,
         proj: nn.Module,
-        kernel_a: nn.Module = GaussianRBF(trainable=True),
-        kernel_b: Optional[nn.Module] = GaussianRBF(trainable=True),
+        kernel_a: Union[nn.Module, str] = 'rbf',
+        kernel_b: Optional[Union[nn.Module, str]] = 'rbf',
```

Comment:
I would say to use `Literal`s but I know you'll point me to the other open issue... :)

Reply:
Ha ha I hope I'm not developing a reputation for deflecting to other issues!

I can see why `Literal` might be a good idea here since we really do only accept `rbf` as a `str`. Since we'd want to do this for the pytorch and tensorflow kernels it might be best for me to follow-up straight after this PR?

Reply:
Tracker: https://github.com/SeldonIO/alibi-detect/issues/683